hq_superset/hq_domain.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+import jinja2
+
+from flask import flash, g, redirect, request, url_for, session
+from flask_login import current_user
+
+def before_request_hook():
+    override_jinja2_template_loader()
+    return ensure_domain_selected()
+
+
+def override_jinja2_template_loader():
+    # Allow loading templates from the templates directory in this project as well
+    from superset import app
+
+    template_path = os.sep.join((
+        os.path.dirname(os.path.abspath(__file__)),
+        'templates'
+    ))
+    my_loader = jinja2.ChoiceLoader([
+            jinja2.FileSystemLoader([template_path]),
+            app.jinja_loader,
+        ])
+    app.jinja_loader = my_loader
+
+
+DOMAIN_EXCLUDED_VIEWS = [
+    "AuthOAuthView.login",
+    "AuthOAuthView.logout",
+    "AuthOAuthView.oauth_authorized",
+    "AuthDBView.logout",
+    "AuthDBView.login",
+    "SelectDomainView.list",
+    "appbuilder.static",
+]
+
+def ensure_domain_selected():
+    # Check if a hq_domain cookie is set
+    #   Ensure necessary roles, permissions and DB schemas are created for the domain
+    import superset
+    if request.url_rule.endpoint in DOMAIN_EXCLUDED_VIEWS:
+        return
+    hq_domain = request.cookies.get('hq_domain')
+    valid_domains = user_domains(current_user)
+    # Todo; Handle no superset enabled domains case
```

Comment:
What's the plan for this? Just redirect to a 404?

Reply:
Actually, I removed this comment, the behaviour is that it keeps going to select-domain page where no domains are available to select.